src/sentry/relay/config/metric_extraction.py

```diff
@@ -102,10 +110,10 @@ def _threshold_to_rules(
             "inner": [
                 {
                     "op": "gt",
-                    "name": _TRANSACTION_METRICS_TO_RULE_FIELD[threshold.metric],
+                    "name": _TRANSACTION_METRICS_TO_RULE_FIELD[cast(int, threshold.metric)],
                     # The frustration threshold is always four times the threshold
                     # (see https://docs.sentry.io/product/performance/metrics/#apdex)
-                    "value": threshold.threshold * 4,
+                    "value": cast(int, threshold.threshold) * 4,
```

Comment:
probably need to annotate the model fields like:

```python
field: WhateverType[int, int] = WhateverType(...)
```

django-stubs has a get type / set type and generics have no way to default to anything but the bounds

Reply:
oh interesting. let me try that

Reply:
hmm can't get it to work. wanna give it a try? I am trying to set it to `field: "models.Field[int, int]"` (because PositiveSmallIntegerField and IntegerField are not subscriptable)

Reply:
which module is it in?

Reply:
It is about _two_ models, ProjectTransactionThreshold and ProjectTransactionThresholdOverride here: https://github.com/getsentry/sentry/blob/ref/relay-typing/src/sentry/models/transaction_threshold.py

`threshold` is a `Union` of a bunch of models that have the same field